simulate_websockets.py

```
@@ -0,0 +1,167 @@
+        print(f"  {RED}✗{NC} #{index+1} {user_id} disconnected: {e}")
+    except ConnectionRefusedError:
+        print(f"  {RED}✗{NC} #{index+1} Connection refused — is the backend running?")
+    except Exception as e:
+        print(f"  {RED}✗{NC} #{index+1} {user_id} error: {e}")
+
+
+async def get_vault_ids() -> list[str]:
+    """Fetch vault IDs from the admin API."""
+    try:
+        import urllib.request
+        admin_token = "sk_TBn8_PzP-CD9BIODfdILDRQqIy_yGAb6kfbvHeanG2g"
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to: "admin_token"

(S105)